### ISSUE TITLE
Ignore unknown keys per class / via annotation

### ISSUE THREAD

**Issue by sschuberth** [Kotlin/kotlinx.serialization#1420]
**What is your use-case and why do you need this feature?**

I know [keys can be ignored "globally"](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/json.md#ignoring-unknown-keys) on the JSON builder scope. But for safely, and to better document which classes have been implemented completely and which not, I'd prefer to ignore unknown keys per class only.

**Describe the solution you'd like**

I'd prefer an annotation-based solution, similar to Jackson's `@JsonIgnoreProperties(ignoreUnknown = true)` annotation.

**Comment by codeyash** [comment]
any update?

Its very critial for scalability.

**Comment by ixtli** [comment]
Yeah there are many major APIs were forced to use which version their API contracts publicly but still randomly permute certain sub objects.
